trunk/SUAVE/Components/Energy/Converters/de_Laval_Nozzle.py

```diff
@@ -0,0 +1,160 @@
+    """
+    
+    def __defaults__(self):
+        """ This sets the default values for the component to function.
+        
+        Assumptions:
+        None
+    
+        Source:
+        N/A
+    
+        Inputs:
+        None
+    
+        Outputs:
+        None
+    
+        Properties Used:
+        None
+        """        
+        
+        #set the defaults
+        self.tag = 'Nozzle'
+        self.polytropic_efficiency           = 1.0
+        self.pressure_ratio_converge         = 1.0
+        self.pressure_ratio_diverge          = 1.0
+        self.expansion_ratio                 = 1.0     
+        self.area_throat                     = 1.0
+        
+    def compute(self,conditions):
+        """This computes the output values from the input values according to
+        equations from the source.
+        
+        Assumptions:
+        Constant polytropic efficiency and pressure ratio
+        Isentropic Process from chmaber to throat (Pt_throat = Pt2)
+        Supersonic/Throat is Choked
+        
+        Source:
+        Chapter 7
+        https://web.stanford.edu/~cantwell/AA283_Course_Material/AA283_Course_Notes/
+        
+        Inputs:
+        self.inputs.
+          combustion_temperature              [k]
+          combustion_pressure                 [Pa]
+          isentropic_expansion_factor         [-]
+          gas_specific_constant               [J/(kg-K)]
+                             
+        Outputs:
+        self.outputs.
+          stagnation_pressure                 [Pa]
+          stagnation_temperature              [K]  
+          static_temperature                  [K]
+          static_pressure                     [Pa]
+          mach_number                         [-]
+          velocity                            [m/s]
+                
+        Properties Used:
+        self.
+          pressure_ratio                      [-]
+          polytropic_efficiency               [-]
+          area_throat                         [m^2]
+          expansion_ratio                     [-]
+        """           
+
+        #--Unpack the values--
+
+        # unpack from inputs
+        Pt_combustion  = self.inputs.combustion_pressure
+        Tt_combustion  = self.inputs.combustion_temperature
+        gamma          = self.inputs.isentropic_expansion_factor
+        R              = self.inputs.gas_specific_constant     
+        
+        # unpack from self
+        pi              = self.pressure_ratio
+        etapold         = self.polytropic_efficiency
+        expansion_ratio = self.expansion_ratio
+        area_throat     = self.area_throat 
+        
+        
+        #--Method for computing the nozzle properties--
+        
+        #--Converging Nozzle --
+        # getting stagnation quantities at throat
+        Pt_throat   = Pt_combustion*pi
+        Tt_throat   = Tt_combustion*pi**((gamma-1.)/(gamma)*etapold)        
```

Comment:
if you expect to run this calculation often you could pre-calculate gamma-1 as gamma_m_ and... so on. But it's not a huge necessity